File: JavaScript/7js 1/arrCheck.js
/*
|--------------------------------------------------------------------------
| Is every value in the array an array?
|--------------------------------------------------------------------------
|
*/

function arrCheck(value) {
   return value.filter((i) => Array.isArray(i)).length == value.length;
}


function arrCheck(value) {
   return value.every(Array.isArray);
}


function arrCheck(arr) {
   return !arr.some((x) => !Array.isArray(x));
}


console.log(arrCheck([]), true);
console.log(arrCheck([["string"]]), true);
console.log(arrCheck([[], {}]), false);
console.log(arrCheck([[1], [2], [3]]), true);
console.log(arrCheck(["A", "R", "R", "A", "Y"]), false);

File: Python/7py 1/arr_check.py
"""
|--------------------------------------------------------------------------
| Is every value in the array an array?
|--------------------------------------------------------------------------
|
"""

def arr_check(arr):
   return len([i for i in arr if type(i)==list]) == len(arr)


def arr_check(arr):
   return all(type(i) == list for i in arr)


def arr_check(arr):
   return all(isinstance(el, list) for el in arr)


print(arr_check([]), True)
print(arr_check([['string']]), True)
print(arr_check([[], {}]), False)
print(arr_check([[1], [2], [3]]), True)
print(arr_check(["A", "R", "R", "A", "Y"]), False)